packages/fast-tooling/src/message-system-service/shortcuts-service-actions/shortcuts.service-action.duplicate.ts

```diff
@@ -1,11 +1,11 @@
-import { ShortcutsAction } from "../shortcuts.service-action";
+import { ShortcutsAction } from "../shortcuts.service-action.js";
 import {
     MessageSystem,
     MessageSystemDataTypeAction,
     MessageSystemType,
-} from "../../message-system";
-import { shortcutsId } from "../shortcuts.service";
-import { AddLinkedDataDataMessageIncoming } from "../../";
+} from "../../message-system/index.js";
+import { shortcutsId } from "../shortcuts.service.js";
+import { AddLinkedDataDataMessageIncoming } from "../../message-system/message-system.utilities.props.js";
```

Comment:
Do these exports not bubble up? Seems like for certain imports you are having to specify the file they exist in rather than the root level index.

Reply:
The imports can be imported from anywhere they're exposed, just like always. You're probably seeing some different patterns because I did a regex find + replace to add the extension and then came back around and patched up anything that was missed manually. Rather than trying to find the exact file from which the export came from I just targed `index.js` because it is there. In general, it is best to be specific with import paths to avoid any side effects and circular dependencies that could result from importing from a more broad module. Additionally, ES doesn't support directory imports (which is all the index.js is) so it is probably best, over time, to phase out the `index.js` convention. I want to make this change as transparently as possible though, so I think that should be a follow-up / refactor.